src/components/dialogue.py

```diff
@@ -0,0 +1,270 @@
+import pygame as pg
+from queue import Queue
+from treelib import Tree
+
+import os
+import sys
+
+#needed to get modules
+parent_dir = os.path.dirname(os.path.dirname(__file__))
+sys.path.append(parent_dir)
+
+
+from modules.state_machine import Machine, DisplayEngine
+#Dialogue Specific Settings
+TEXT_SPEED = 0.03
+
+class _DialogueStateMachine(Machine):
+  def __init__(self):
+    self.current : DialogueState = None
+    self.next_state : DialogueState = None
+
+
+  def update(self):
+    try:
+      if self.next_state:
+        self.current = self.next_state
+        self.next_state = None
+    except:
+      print("ERROR SWITCHING")
+      pass
+
+
+#represents Dialogue surface drawing logic int he game...
+class Dialogue():
+  def __init__(self, text : str = "This is just an example text to use with gradual typing."):
+    self.text = text
+    self.surf = pg.Surface((700,200), pg.SRCALPHA)
+    self.show_textbox = False
+    self.typing = False
+    self.surf.fill((0, 0, 255, 100))
+    self.rendering = ''
+    self.FONT = pg.font.SysFont(None, 24, 0)
+    self.textbox_rect = self.surf.get_rect(topleft=(150,200))
+    self.border_rect = self.surf.get_rect(topleft=(0, 0))
+    self.index = 0
+
+    self.time_delay = TEXT_SPEED #set 3 to a TIME_CONSTANT in settings.py
+
+  def draw_border(self):
+    self.surf.fill((0, 0, 255, 100))
+    pg.draw.rect(self.surf, "Black", self.border_rect, 6)
+
+  @property
+  def is_done(self):
+    if self.index == len(self.text) - 1:
+      return True
+    else:
+      return False
+
+  def draw(self, surface):
+    self.draw_border()
+
+    if self.index < len(self.text) - 1:
+        #slows down the time of game..
+        rendered_text = self.FONT.render(self.rendering, 1, 'White')
+        text_rect = rendered_text.get_rect(topleft=(20, 90))
+
+        self.surf.blit(rendered_text, text_rect)
+        surface.blit(self.surf, self.textbox_rect)
+        pg.display.flip()
+    else:
+        rendered_text = self.FONT.render(self.text, 1, 'White')
+        text_rect = rendered_text.get_rect(topleft=(20, 90))
+        self.surf.blit(rendered_text, text_rect)
+        surface.blit(self.surf, self.textbox_rect)
+        pg.display.flip()
```

Comment:
It slows down by the update function through delta time as it subtracts the delay by 0.03 of a second. Just know that you can change the setting TEXT_SPEED at the top and it will automatically adjust. Probably can move it settings.py or make it a variable inside of the state.

Yeah, gimme a day and I'll clean up the code. Got me stressin last night when my branch disappeared out of thin air. Had to pr request to main incase I lost my work 😭

Reply:
Your branch disappeared? do you know what might've caused it? O_O

Reply:
> Your branch disappeared? do you know what might've caused it? O_O

lmao yeah my head was detached so I was building off a commit not connected to head. I just git reflog and tagged it. Then merged into my new branch name and changed a few things thened PR.